qcodes/utils/delaykeyboardinterrupt.py

```diff
@@ -0,0 +1,41 @@
+import signal
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class DelayedKeyboardInterrupt:
+    """
+    A context manager to wrap a piece of code to ensure that a KeyboardInterrupt is not
+    triggered by a SIGINT during the execution of this context. A second SIGINT will trigger the
+    KeyboardInterrupt immediately.
+
+    Inspired by https://stackoverflow.com/questions/842557/how-to-prevent-a-block-of-code-from-being-interrupted-by-keyboardinterrupt-in-py
+    """
+    signal_received = None
+    old_handler = None
+
+    def __enter__(self) -> None:
+        if signal.getsignal(signal.SIGINT) is signal.default_int_handler:
+            self.old_handler = signal.signal(signal.SIGINT, self.handler)
```

Comment:
which `self.handler` is meant here? perhaps `self.old_handler`?
```suggestion
            self.old_handler = signal.signal(signal.SIGINT, self.old_handler)
```